src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -371,6 +373,11 @@ public void Initialize(ITelemetry telemetry)
             if (string.IsNullOrEmpty(instrumentationKey))
             {
                 instrumentationKey = this.configuration.InstrumentationKey;
+
+                if (string.IsNullOrEmpty(instrumentationKey))
+                {
+                    instrumentationKey = PlatformSingleton.Current.GetEnvironmentVariable(InstrumentationKeyWebSitesEnvironmentVariable);
```

Comment:
So this ikey has the least priority? Is it what we want for Web Apps case? In Web Apps you can configure ikey in `AI.config` file and then override it using Web App settings. Not other way around.

Also I'm concerned about performance of it. Maybe just read it once and cache? Or we need to support the scenario when environment variable will be changed later?